netdata/packaging/installer: User management during install/uninstall broken for MacOS

##### Bug report summary
When running the installer on Mac, user management of the installer is broken.
We need to adopt the mac flavour of commands for adding/removing users.

Indicative article with some basic instructions:
https://apple.stackexchange.com/questions/226073/how-do-i-create-user-accounts-from-the-terminal-in-mac-os-x-10-11

**Workaround:** The user can manually create the necessary groups and respectively delete them upon uninstall

We can prioritise this if we find the community having major problems with this

##### OS / Environment
MacOS

##### Netdata version (ouput of `netdata -V`)
latest
##### Component Name
netdata/packaging/installer

##### Steps To Reproduce
Just run the installer on Mac following the netdata instructions for Mac

##### Expected behavior
We should properly create/modify/delete users and groups as needed in the install/uninstall process.

So i went ahead and re-run the scenarios to confirm the findings one more time.

### 1st scenario: Running the installer as non-root and with --install /tmp/netdata

The errors/warnings seen are:

```
$ ./netdata-installer.sh --install /tmp/netdata

  IMPORTANT:
  You are about to install netdata as a non-root user.
  Netdata will work, but a few data collection modules that
  require root access will fail.

  If you installing netdata permanently on your system, run
  the installer like this:

     sudo ./netdata-installer.sh 

```

```
 --- Add user netdata to required user groups --- 
 FAILED  The installer does not run as root. 
```

and

```
 --- Install netdata at system init --- 
 FAILED  Cannot install netdata init service. 
```


### 2nd Scenario: Run non-root install without --install option
```
$ ./netdata-installer.sh 

  ^
  |.-.   .-.   .-.   .-.   .-.   .  netdata                      .   .-.   .-
  |   '-'   '-'   '-'   '-'   '-'   wrong command line options!   '-'   '-'  
  +----+-----+-----+-----+-----+-----+-----+-----+-----+-----+-----+-----+--->


  Sorry! This will fail!

  You are attempting to install netdata as non-root, but you plan
  to install it in system paths.

  Please set an installation prefix, like this:

      ./netdata-installer.sh  --install /tmp

  or, run the installer as root:

      sudo ./netdata-installer.sh 

  We suggest to install it as root, or certain data collectors will
  not be able to work. Netdata drops root privileges when running.
  So, if you plan to keep it, install it as root to get the full
  functionality.
```

### 3rd scenario: Run the installer with sudo and with --install option
The install went fine, except that it didn't mention process for adding a new user and also whatever included user/group permission it popped errors like the following

```
[/Users/paulkats/Documents/Development/Personal/netdata]# chown root:netdata /tmp/netdata/netdata/usr/libexec/netdata/plugins.d/go.d.plugin 
chown: netdata: illegal group name
 FAILED   
```

### 4rth scenario: Run the installer with sudo, without --install option
Install fails epically with the following error

```
 --- Backup existing netdata configuration before installing it --- 
 --- Install netdata --- 
[/Users/paulkats/Documents/Development/Personal/netdata]# make install 
Making install in diagrams
make[2]: Nothing to be done for `install-exec-am'.
make[2]: Nothing to be done for `install-data-am'.
Making install in system
make[2]: Nothing to be done for `install-exec-am'.
 /usr/local/bin/gmkdir -p '/etc/netdata'
 /usr/local/bin/ginstall -c edit-config '/etc/netdata'
Making install in tests
make[2]: Nothing to be done for `install-exec-am'.
 /usr/local/bin/gmkdir -p '/usr/libexec/netdata/plugins.d'
gmkdir: cannot create directory '/usr/libexec/netdata': Operation not permitted
make[2]: *** [install-dist_pluginsSCRIPTS] Error 1
make[1]: *** [install-am] Error 2
make: *** [install-recursive] Error 1
 FAILED   
```
```

Following the instructions here: https://docs.netdata.cloud/packaging/installer/#macos


I notice that the Homebrew approach works just fine for my mac

```
p@XXX:~/on  fix-installer-macos
$ brew info netdata
netdata: stable 1.14.0 (bottled)
Distributed real-time performance and health monitoring
https://my-netdata.io/
Not installed
From: https://github.com/Homebrew/homebrew-core/blob/master/Formula/netdata.rb
==> Dependencies
Build: autoconf ✔, automake ✔, pkg-config ✔
==> Caveats
To have launchd start netdata now and restart at login:
  brew services start netdata
Or, if you don't want/need a background service you can just run:
  /usr/local/sbin/netdata -D
==> Analytics
install: 648 (30 days), 1,465 (90 days), 3,596 (365 days)
install_on_request: 646 (30 days), 1,458 (90 days), 3,590 (365 days)
build_error: 0 (30 days)

p@XXX:~/ on  fix-installer-macos
$ brew install netdata
==> Downloading https://homebrew.bintray.com/bottles/netdata-1.14.0_1.mojave.bottle.tar.gz
==> Downloading from https://akamai.bintray.com/41/41b7ad105a34c6d08504b519fdd831083f2e5eafae50e56dbb5f2764bf42fb3e?__gda__=exp=1558102885~hmac=ed7e00a59fa0c0b55c4553618e42129d98356bd3771455dea255
######################################################################## 100.0%
==> Pouring netdata-1.14.0_1.mojave.bottle.tar.gz
==> Caveats
To have launchd start netdata now and restart at login:
  brew services start netdata
Or, if you don't want/need a background service you can just run:
  /usr/local/sbin/netdata -D
==> Summary
🍺  /usr/local/Cellar/netdata/1.14.0_1: 457 files, 6.7MB

p@XXX:~/ on  fix-installer-macos
```


But when i try to build from source, compilation breaks:
```
  CC       registry/registry_url.o
  CC       database/rrdcalc.o
  CC       database/rrdcalctemplate.o
  CC       database/rrddim.o
  CC       database/rrddimvar.o
  CC       database/rrdfamily.o
  CC       database/rrdhost.o
  CC       database/rrd.o
  CC       database/rrdset.o
  CC       database/rrdsetvar.o
  CC       database/rrdvar.o
  CC       streaming/rrdpush.o
  CC       collectors/statsd.plugin/statsd.o
  CC       web/server/web_client.o
database/rrdset.c:1265:168: warning: format specifies type 'long' but the argument has type '__darwin_suseconds_t' (aka 'int') [-Wformat]
  ...too old data (last updated at %ld.%ld, last collected at %ld.%ld). Resetting it. Will not store the next entry.", st->name, st->last_updated.tv_sec, st->last_updated.tv_usec, st->last_coll...
                                       ~~~                                                                                                                ^~~~~~~~~~~~~~~~~~~~~~~~
                                       %d
database/../daemon/../libnetdata/log/log.h:85:71: note: expanded from macro 'info'
#define info(args...)    info_int(__FILE__, __FUNCTION__, __LINE__, ##args)
                                                                      ^~~~
database/rrdset.c:1265:226: warning: format specifies type 'long' but the argument has type '__darwin_suseconds_t' (aka 'int') [-Wformat]
  ...at %ld.%ld). Resetting it. Will not store the next entry.", st->name, st->last_updated.tv_sec, st->last_updated.tv_usec, st->last_collected_time.tv_sec, st->last_collected_time.tv_usec);
            ~~~                                                                                                                                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
            %d
database/../daemon/../libnetdata/log/log.h:85:71: note: expanded from macro 'info'
#define info(args...)    info_int(__FILE__, __FUNCTION__, __LINE__, ##args)
                                                                      ^~~~
  CC       web/server/web_server.o
  CC       web/server/web_client_cache.o
  CC       web/server/static/static-threaded.o
  CC       collectors/macos.plugin/plugin_macos.o
2 warnings generated.
  CC       collectors/macos.plugin/macos_sysctl.o
  CC       collectors/macos.plugin/macos_mach_smi.o
  CC       collectors/macos.plugin/macos_fw.o
  CCLD     cups.plugin
collectors/macos.plugin/macos_fw.c:118:35: warning: address of array 'diskstat.name' will always evaluate to 'true' [-Wpointer-bool-conversion]
            if(unlikely(!diskstat.name || !*diskstat.name)) {
                        ~~~~~~~~~~^~~~
./config.h:345:41: note: expanded from macro 'unlikely'
#define unlikely(x) __builtin_expect(!!(x), 0)
                                        ^
1 warning generated.
  GEN      netdata
undef: ___CFConstantStringClassReference
undef: _kCFAllocatorDefault
undef: _IOIteratorReset
undef: _CFNumberGetValue
undef: _IOServiceMatching
undef: _IOServiceGetMatchingServices
undef: _IORegistryEntryGetChildEntry
undef: _IOIteratorNext
undef: _CFStringGetCString
undef: _IORegistryEntryCreateCFProperties
undef: _CFDictionaryGetValue
undef: _IOMasterPort
undef: _CFRelease
undef: _IOObjectRelease
Undefined symbols for architecture x86_64:
  "___CFConstantStringClassReference", referenced from:
      CFString in lto.o
      CFString in lto.o
      CFString in lto.o
      CFString in lto.o
      CFString in lto.o
      CFString in lto.o
      CFString in lto.o
      ...
  "_kCFAllocatorDefault", referenced from:
      _macos_main in lto.o
  "_IOIteratorReset", referenced from:
      _macos_main in lto.o
  "_CFNumberGetValue", referenced from:
      _macos_main in lto.o
  "_IOServiceMatching", referenced from:
      _macos_main in lto.o
  "_IOServiceGetMatchingServices", referenced from:
      _macos_main in lto.o
  "_IORegistryEntryGetChildEntry", referenced from:
      _macos_main in lto.o
  "_IOIteratorNext", referenced from:
      _macos_main in lto.o
  "_CFStringGetCString", referenced from:
      _macos_main in lto.o
  "_IORegistryEntryCreateCFProperties", referenced from:
      _macos_main in lto.o
  "_CFDictionaryGetValue", referenced from:
      _macos_main in lto.o
  "_IOMasterPort", referenced from:
      _macos_main in lto.o
  "_CFRelease", referenced from:
      _macos_main in lto.o
  "_IOObjectRelease", referenced from:
      _macos_main in lto.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [netdata] Error 1
make[1]: *** [all-recursive] Error 1
make: *** [all] Error 2
 FAILED   
```

--- a/netdata-installer.sh
+++ b/netdata-installer.sh
@@ -515,20 +515,24 @@ progress "Fix generated files permissions"
 run find ./system/ -type f -a \! -name \*.in -a \! -name Makefile\* -a \! -name \*.conf -a \! -name \*.service -a \! -name \*.logrotate -exec chmod 755 {} \;
 
 # -----------------------------------------------------------------------------
-progress "Add user netdata to required user groups"
+progress "Creating standard user and groups for netdata"
 
 NETDATA_WANTED_GROUPS="docker nginx varnish haproxy adm nsd proxy squid ceph nobody"
 NETDATA_ADDED_TO_GROUPS=""
 if [ "${UID}" -eq 0 ]; then
+	progress "Adding group 'netdata'"
 	portable_add_group netdata || :
+
+	progress "Adding user 'netdata'"
 	portable_add_user netdata "${NETDATA_PREFIX}/var/lib/netdata" || :
 
+	progress "Assign user 'netdata' to required groups"
 	for g in ${NETDATA_WANTED_GROUPS}; do
 		# shellcheck disable=SC2086
 		portable_add_user_to_group ${g} netdata && NETDATA_ADDED_TO_GROUPS="${NETDATA_ADDED_TO_GROUPS} ${g}"
 	done
 else
-	run_failed "The installer does not run as root."
+	run_failed "The installer does not run as root. Nothing to do for user and groups"
 fi
 
 # -----------------------------------------------------------------------------
@@ -567,6 +571,7 @@ else
 fi
 NETDATA_GROUP="$(id -g -n "${NETDATA_USER}")"
 [ -z "${NETDATA_GROUP}" ] && NETDATA_GROUP="${NETDATA_USER}"
+echo >&2 "Netdata user and group is finally set to: ${NETDATA_USER}/${NETDATA_GROUP}"
 
 # the owners of the web files
 NETDATA_WEB_USER="$(config_option "web" "web files owner" "${NETDATA_USER}")"

--- a/packaging/installer/functions.sh
+++ b/packaging/installer/functions.sh
@@ -606,6 +606,11 @@ portable_add_user() {
 		run adduser -h "${homedir}" -s "${nologin}" -D -G "${username}" "${username}" && return 0
 	fi
 
+	# mac OS
+	if command -v sysadminctl 1> /dev/null 2>&1; then
+		run sysadminctl -addUser  ${username} && return 0
+	fi
+
 	echo >&2 "Failed to add ${username} user account !"
 
 	return 1
@@ -635,6 +640,11 @@ portable_add_group() {
 	# BusyBox
 	if command -v addgroup 1>/dev/null 2>&1; then
 		run addgroup "${groupname}" && return 0
+	fi
+
+	# mac OS
+	if command -v dseditgroup 1> /dev/null 2>&1; then
+		dseditgroup -o create "${groupname}" && return 0
 	fi
 
 	echo >&2 "Failed to add ${groupname} user group !"
@@ -674,6 +684,10 @@ portable_add_user_to_group() {
 			run addgroup "${username}" "${groupname}" && return 0
 		fi
 
+		# mac OS
+		if command -v dseditgroup 1> /dev/null 2>&1; then
+			dseditgroup -u "${username}" "${groupname}" && return 0
+		fi
 		echo >&2 "Failed to add user ${username} to group ${groupname} !"
 		return 1
 	fi

--- a/packaging/installer/netdata-uninstaller.sh
+++ b/packaging/installer/netdata-uninstaller.sh
@@ -1,11 +1,15 @@
 #!/usr/bin/env bash
 #shellcheck disable=SC2181
-
-# this script will uninstall netdata
-
+#
+# This is the netdata uninstaller script
 # Variables needed by script and taken from '.environment' file:
 #  - NETDATA_PREFIX
 #  - NETDATA_ADDED_TO_GROUPS
+#
+# Copyright: SPDX-License-Identifier: GPL-3.0-or-later
+#
+# Author: Paul Emm. Katsoulakis <paul@netdata.cloud>
+#
 
 usage="$(basename "$0") [-h] [-f ] -- program to calculate the answer to life, the universe and everything
 
@@ -46,13 +50,13 @@ while :; do
 done
 
 if [ "$YES" != "1" ]; then
-	echo "This script will REMOVE netdata from your system."
-	echo "Run it again with --yes to do it."
+	echo >&2 "This script will REMOVE netdata from your system."
+	echo >&2 "Run it again with --yes to do it."
 	exit 1
 fi
 
 if [[ $EUID -ne 0 ]]; then
-	echo "This script SHOULD be run as root or otherwise it won't delete all installed components."
+	echo >&2 "This script SHOULD be run as root or otherwise it won't delete all installed components."
 	key="n"
 	read -r -s -n 1 -p "Do you want to continue as non-root user [y/n] ? " key
 	if [ "$key" != "y" ] && [ "$key" != "Y" ]; then
@@ -60,34 +64,227 @@ if [[ $EUID -ne 0 ]]; then
 	fi
 fi
 
-function quit_msg() {
+# -----------------------------------------------------------------------------
+
+setup_terminal() {
+	TPUT_RESET=""
+	TPUT_BLACK=""
+	TPUT_RED=""
+	TPUT_GREEN=""
+	TPUT_YELLOW=""
+	TPUT_BLUE=""
+	TPUT_PURPLE=""
+	TPUT_CYAN=""
+	TPUT_WHITE=""
+	TPUT_BGBLACK=""
+	TPUT_BGRED=""
+	TPUT_BGGREEN=""
+	TPUT_BGYELLOW=""
+	TPUT_BGBLUE=""
+	TPUT_BGPURPLE=""
+	TPUT_BGCYAN=""
+	TPUT_BGWHITE=""
+	TPUT_BOLD=""
+	TPUT_DIM=""
+	TPUT_UNDERLINED=""
+	TPUT_BLINK=""
+	TPUT_INVERTED=""
+	TPUT_STANDOUT=""
+	TPUT_BELL=""
+	TPUT_CLEAR=""
+
+	# Is stderr on the terminal? If not, then fail
+	test -t 2 || return 1
+
+	if command -v tput 1>/dev/null 2>&1; then
+		if [ $(($(tput colors 2>/dev/null))) -ge 8 ]; then
+			# Enable colors
+			TPUT_RESET="$(tput sgr 0)"
+			TPUT_BLACK="$(tput setaf 0)"
+			TPUT_RED="$(tput setaf 1)"
+			TPUT_GREEN="$(tput setaf 2)"
+			TPUT_YELLOW="$(tput setaf 3)"
+			TPUT_BLUE="$(tput setaf 4)"
+			TPUT_PURPLE="$(tput setaf 5)"
+			TPUT_CYAN="$(tput setaf 6)"
+			TPUT_WHITE="$(tput setaf 7)"
+			TPUT_BGBLACK="$(tput setab 0)"
+			TPUT_BGRED="$(tput setab 1)"
+			TPUT_BGGREEN="$(tput setab 2)"
+			TPUT_BGYELLOW="$(tput setab 3)"
+			TPUT_BGBLUE="$(tput setab 4)"
+			TPUT_BGPURPLE="$(tput setab 5)"
+			TPUT_BGCYAN="$(tput setab 6)"
+			TPUT_BGWHITE="$(tput setab 7)"
+			TPUT_BOLD="$(tput bold)"
+			TPUT_DIM="$(tput dim)"
+			TPUT_UNDERLINED="$(tput smul)"
+			TPUT_BLINK="$(tput blink)"
+			TPUT_INVERTED="$(tput rev)"
+			TPUT_STANDOUT="$(tput smso)"
+			TPUT_BELL="$(tput bel)"
+			TPUT_CLEAR="$(tput clear)"
+		fi
+	fi
+
+	return 0
+}
+setup_terminal || echo >/dev/null
+
+run_ok() {
+	printf >&2 "${TPUT_BGGREEN}${TPUT_WHITE}${TPUT_BOLD} OK ${TPUT_RESET} ${*} \n\n"
+}
+
+run_failed() {
+	printf >&2 "${TPUT_BGRED}${TPUT_WHITE}${TPUT_BOLD} FAILED ${TPUT_RESET} ${*} \n\n"
+}
+
+ESCAPED_PRINT_METHOD=
+printf "%q " test >/dev/null 2>&1
+[ $? -eq 0 ] && ESCAPED_PRINT_METHOD="printfq"
+escaped_print() {
+	if [ "${ESCAPED_PRINT_METHOD}" = "printfq" ]; then
+		printf "%q " "${@}"
+	else
+		printf "%s" "${*}"
+	fi
+	return 0
+}
+
+run_logfile="/dev/null"
+run() {
+	local user="${USER--}" dir="${PWD}" info info_console
+
+	if [ "${UID}" = "0" ]; then
+		info="[root ${dir}]# "
+		info_console="[${TPUT_DIM}${dir}${TPUT_RESET}]# "
+	else
+		info="[${user} ${dir}]$ "
+		info_console="[${TPUT_DIM}${dir}${TPUT_RESET}]$ "
+	fi
+
+	printf >>"${run_logfile}" "${info}"
+	escaped_print >>"${run_logfile}" "${@}"
+	printf >>"${run_logfile}" " ... "
+
+	printf >&2 "${info_console}${TPUT_BOLD}${TPUT_YELLOW}"
+	escaped_print >&2 "${@}"
+	printf >&2 "${TPUT_RESET}\n"
+
+	"${@}"
+
+	local ret=$?
+	if [ ${ret} -ne 0 ]; then
+		run_failed
+		printf >>"${run_logfile}" "FAILED with exit code ${ret}\n"
+	else
+		run_ok
+		printf >>"${run_logfile}" "OK\n"
+	fi
+
+	return ${ret}
+}
+
+portable_del_group() {
+	local groupname="${1}"
+
+	# Check if group exist
+	echo >&2 "Removing ${groupname} user group ..."
+
+	# Linux
+	if command -v groupdel 1>/dev/null 2>&1; then
+		run groupdel -f "${groupname}" && return 0
+	fi
+
+	# mac OS
+	if command -v dseditgroup 1> /dev/null 2>&1; then
+		if dseditgroup -o read netdata 1> /dev/null 2>&1; then
+			run dseditgroup -o delete "${groupname}" && return 0
+		else
+			echo >&2 "Could not find group ${groupname}, nothing to do"
+		fi
+	fi
+
+	echo >&2 "Group ${groupname} was not automatically removed, you might have to remove it manually"
+	return 1
+}
+
+portable_del_user() {
+	local username="${1}"
+	echo >&2 "Deleting ${username} user account ..."
+
+	# Linux
+	if command -v userdel 1>/dev/null 2>&1; then
+		run userdel -f "${username}" && return 0
+	fi
+
+	# mac OS
+	if command -v sysadminctl 1>/dev/null 2>&1; then
+		run sysadminctl -deleteUser "${username}" && return 0
+	fi
+
+	echo >&2 "User ${username} could not be deleted from system, you might have to remove it manually"
+	return 1
+}
+
+portable_del_user_from_group() {
+	local groupname="${1}" username="${2}"
+
+	# username is not in group
+	echo >&2 "Deleting ${username} user from ${groupname} group ..."
+
+	# Linux
+	if command -v gpasswd 1>/dev/null 2>&1; then
+		run gpasswd -d "netdata" "${group}" && return 0
+	fi
+
+	# FreeBSD
+	if command -v pw 1>/dev/null 2>&1; then
+		run pw groupmod "${groupname}" -d "${username}" && return 0
+	fi
+
+	# BusyBox
+	if command -v delgroup 1>/dev/null 2>&1; then
+		run delgroup "${username}" "${groupname}" && return 0
+	fi
+
+	# mac OS
+	if command -v dseditgroup 1> /dev/null 2>&1; then
+		run dseditgroup -o delete -u "${username}" "${groupname}" && return 0
+	fi
+
+	echo >&2 "Failed to delete user ${username} from group ${groupname} !"
+	return 1
+}
+
+quit_msg() {
 	echo
 	if [ "$FILE_REMOVAL_STATUS" -eq 0 ]; then
-		echo "Something went wrong :("
+		echo >&2 "Something went wrong :("
 	else
-		echo "Netdata files were successfully removed from your system"
+		echo >&2 "Netdata files were successfully removed from your system"
 	fi
 }
 
-function user_input() {
+user_input() {
 	TEXT="$1"
 	if [ "${INTERACTIVITY}" == "-i" ]; then
 		read -r -p "$TEXT" >&2
 	fi
 }
 
-function rm_file() {
+rm_file() {
 	FILE="$1"
 	if [ -f "${FILE}" ]; then
-		rm -v ${INTERACTIVITY} "${FILE}"
+		run rm -v ${INTERACTIVITY} "${FILE}"
 	fi
 }
 
-function rm_dir() {
+rm_dir() {
 	DIR="$1"
 	if [ -n "$DIR" ] && [ -d "$DIR" ]; then
 		user_input "Press ENTER to recursively delete directory '$DIR' > "
-		rm -v -f -R "${DIR}"
+		run rm -v -f -R "${DIR}"
 	fi
 }
 
@@ -116,14 +313,14 @@ trap quit_msg EXIT
 source "${ENVIRONMENT_FILE}" || exit 1
 
 #### STOP NETDATA
-echo "Stopping a possibly running netdata..."
+echo >&2 "Stopping a possibly running netdata..."
 for p in $(netdata_pids); do
 	i=0
 	while kill "${p}" 2>/dev/null; do
 		if [ "$i" -gt 30 ]; then
-			echo "Forcefully stopping netdata with pid ${p}"
-			kill -9 "${p}"
-			sleep 2
+			echo >&2 "Forcefully stopping netdata with pid ${p}"
+			run kill -9 "${p}"
+			run sleep 2
 			break
 		fi
 		sleep 1
@@ -155,15 +352,18 @@ fi
 
 FILE_REMOVAL_STATUS=1
 
-#### REMOVE NETDATA USER & GROUP
+#### REMOVE NETDATA USER FROM ADDED GROUPS
 if [ -n "$NETDATA_ADDED_TO_GROUPS" ]; then
 	user_input "Press ENTER to delete 'netdata' from following groups: '$NETDATA_ADDED_TO_GROUPS' > "
 	for group in $NETDATA_ADDED_TO_GROUPS; do
-		gpasswd -d netdata "${group}"
+		portable_del_user_from_group "${group}" "netdata"
 	done
 fi
 
+#### REMOVE USER
 user_input "Press ENTER to delete 'netdata' system user > "
-userdel -f netdata || :
+portable_del_user "netdata" || :
+
+### REMOVE GROUP
 user_input "Press ENTER to delete 'netdata' system group > "
-groupdel -f netdata || :
+portable_del_group "netdata" || :

--- a/packaging/installer/netdata-uninstaller.sh
+++ b/packaging/installer/netdata-uninstaller.sh
@@ -68,30 +68,12 @@ fi
 
 setup_terminal() {
 	TPUT_RESET=""
-	TPUT_BLACK=""
-	TPUT_RED=""
-	TPUT_GREEN=""
 	TPUT_YELLOW=""
-	TPUT_BLUE=""
-	TPUT_PURPLE=""
-	TPUT_CYAN=""
 	TPUT_WHITE=""
-	TPUT_BGBLACK=""
 	TPUT_BGRED=""
 	TPUT_BGGREEN=""
-	TPUT_BGYELLOW=""
-	TPUT_BGBLUE=""
-	TPUT_BGPURPLE=""
-	TPUT_BGCYAN=""
-	TPUT_BGWHITE=""
 	TPUT_BOLD=""
 	TPUT_DIM=""
-	TPUT_UNDERLINED=""
-	TPUT_BLINK=""
-	TPUT_INVERTED=""
-	TPUT_STANDOUT=""
-	TPUT_BELL=""
-	TPUT_CLEAR=""
 
 	# Is stderr on the terminal? If not, then fail
 	test -t 2 || return 1
@@ -100,30 +82,12 @@ setup_terminal() {
 		if [ $(($(tput colors 2>/dev/null))) -ge 8 ]; then
 			# Enable colors
 			TPUT_RESET="$(tput sgr 0)"
-			TPUT_BLACK="$(tput setaf 0)"
-			TPUT_RED="$(tput setaf 1)"
-			TPUT_GREEN="$(tput setaf 2)"
 			TPUT_YELLOW="$(tput setaf 3)"
-			TPUT_BLUE="$(tput setaf 4)"
-			TPUT_PURPLE="$(tput setaf 5)"
-			TPUT_CYAN="$(tput setaf 6)"
 			TPUT_WHITE="$(tput setaf 7)"
-			TPUT_BGBLACK="$(tput setab 0)"
 			TPUT_BGRED="$(tput setab 1)"
 			TPUT_BGGREEN="$(tput setab 2)"
-			TPUT_BGYELLOW="$(tput setab 3)"
-			TPUT_BGBLUE="$(tput setab 4)"
-			TPUT_BGPURPLE="$(tput setab 5)"
-			TPUT_BGCYAN="$(tput setab 6)"
-			TPUT_BGWHITE="$(tput setab 7)"
 			TPUT_BOLD="$(tput bold)"
 			TPUT_DIM="$(tput dim)"
-			TPUT_UNDERLINED="$(tput smul)"
-			TPUT_BLINK="$(tput blink)"
-			TPUT_INVERTED="$(tput rev)"
-			TPUT_STANDOUT="$(tput smso)"
-			TPUT_BELL="$(tput bel)"
-			TPUT_CLEAR="$(tput clear)"
 		fi
 	fi
 

